apps/prairielearn/src/lib/editors.ts

```diff
@@ -193,30 +201,24 @@ export abstract class Editor {
   protected authz_data: Record<string, any>;
   protected course: Course;
   protected user: User;
-  protected description: string;
+  public readonly description: string;
 
-  constructor(params: BaseEditorOptions) {
+  protected constructor(params: BaseEditorOptionsInternal) {
     this.authz_data = params.locals.authz_data;
     this.course = params.locals.course;
     this.user = params.locals.user;
+    this.description = params.description;
   }
 
   /**
    * Write changes to disk. Returns an object with that paths that `git` should
    * add and the commit message that should be used.
+   *
+   * If no files were changed, return null.
    */
-  protected abstract write(): Promise<WriteResult>;
-
-  /**
-   * Determines whether or not the edit should be executed. For instance, this
-   * can check if the edit would actually modify a file and skip the write/commit/push
-   * steps if it would not.
-   */
-  protected async shouldEdit(): Promise<boolean> {
-    return true;
-  }
+  abstract write(): Promise<WriteResult | null>;
```

Comment:
Having `write()` itself determine if it needs to make an edit ended up being a bit easier to work with in `MultiEditor` and in how the editor is used when it's actually run.